provider/internal/java/util.go

```diff
@@ -147,7 +147,8 @@ func decompile(ctx context.Context, log logr.Logger, archivePath, projectPath st
 			} else {
 				log.V(8).Info("decompiled file", "file", filePath)
 			}
-		} else if strings.HasSuffix(f.Name, JavaArchive) || strings.HasSuffix(f.Name, WebArchive) {
+		// decompile web archives
+		case strings.HasSuffix(f.Name, WebArchive):
```

Comment:
@shawn-hurley only the first level Java archives get decompiled, any JARs inside another JAR mean that they are dependencies, we should only be looking at class files only in the first level to avoid decompiling unnecessary everything

Reply:
https://github.com/konveyor/analyzer-lsp/pull/308/files#diff-072c8024b8b50a75b9f72c30409a7583ef831ebb94cf73b89f209f2282c85356L194-L195

Reply:
heard, this make sense, I don't have the full code mapped in my head anymore so just wanted to double check!